### PR TITLE
Removing unnecessary clusters from tv-casting

### PR DESCRIPTION
**Pull request by cliffamzn**
Removing some clusters that aren't necessary from the tv-casting app.

Also disabling `CHIP_DEVICE_CONFIG_ENABLE_TEST_SETUP_PARAMS` since it
can be dangerous to include in production use cases (see
`config/standalone/CHIPProjectConfig.h`)

### CODE REVIEW

File: src/platform/android/CommissionableDataProviderImpl.cpp [hunk 1]
```diff
@@ -34,6 +34,10 @@ using namespace chip::Crypto;
 
 namespace {
 
+#ifndef CHIP_DEVICE_CONFIG_USE_TEST_SPAKE2P_ITERATION_COUNT
+#define CHIP_DEVICE_CONFIG_USE_TEST_SPAKE2P_ITERATION_COUNT 1000
```
Comment (sharadb-amazon) [inline src/platform/android/CommissionableDataProviderImpl.cpp:38]: Why did this need to be defined here?

File: examples/tv-casting-app/tv-casting-common/include/CHIPProjectAppConfig.h [hunk 2]
```diff
@@ -61,3 +57,12 @@
...
 #define CHIP_DEVICE_CONFIG_ROTATING_DEVICE_ID_UNIQUE_ID_LENGTH 128
+
+// Disable this since it should not be enabled for production setups
+#define CHIP_DEVICE_CONFIG_ENABLE_TEST_SETUP_PARAMS 0
+
+#define CHIP_DEVICE_CONFIG_DYNAMIC_ENDPOINT_COUNT 4
+
+// Include the CHIPProjectConfig from config/standalone
+// Add this at the end so that we can hit our #defines first
+#include <CHIPProjectConfig.h>
```
Comment (sharadb-amazon) [inline examples/tv-casting-app/tv-casting-common/include/CHIPProjectAppConfig.h:68]: I think we were already following the pattern of defining a macro only if it is not defined in this file? Why do we still need to put this at the bottom? I'd rather not rely on the relative ordering of the defines here. If there are some cases that don't follow the right pattern, I'd much rather have them fixed.

**Review by sharadb-amazon — COMMENTED** [review]
The ZAP generation on CI is failing, complaining about a diff: https://github.com/sharadb-amazon/connectedhomeip/actions/runs/4200704183/jobs/7286995427

That needs to succeed if we want to merge this into project-chip/master

File: examples/tv-casting-app/tv-casting-common/include/CHIPProjectAppConfig.h (hunk 2 above)
Comment (cliffamzn) [reply examples/tv-casting-app/tv-casting-common/include/CHIPProjectAppConfig.h:68]: The same configs are defined in `#CHIPProjectConfig.h` these don't get respected, we can either remove the #include altogether (that might cause us to miss stuff added later on)

The `CHIP_DEVICE_CONFIG_ENABLE_TEST_SETUP_PARAMS` should be set and by default it is set to 1

File: src/platform/android/CommissionableDataProviderImpl.cpp (hunk 1 above)
Comment (cliffamzn) [reply src/platform/android/CommissionableDataProviderImpl.cpp:38]: Ugh its annoying, its defined in `CHIPProjectConfig.h` only if  `CHIP_DEVICE_CONFIG_ENABLE_TEST_SETUP_PARAMS` is set to 1

**Review by sharadb-amazon — APPROVED** [review]

**Pull request merged**